packages/alfa-style/src/style.ts

```diff
@@ -101,38 +115,41 @@ export class Style implements Serializable {
     return this._parent.map((parent) => parent.root()).getOr(this);
   }
 
-  public initial<N extends Name>(name: N): Value<Style.Initial<N>>;
-  public initial<N extends Name>(name: N): Value {
-    const property = Property.get(name);
-
-    return Value.of(property.initial);
+  public cascaded<N extends Name>(name: N): Option<Value<Style.Cascaded<N>>> {
+    return Option.from(
+      this._cascaded.get(name) as Value<Style.Cascaded<N>> | undefined
+    );
   }
 
-  public cascaded<N extends Name>(name: N): Option<Value<Style.Cascaded<N>>>;
-  public cascaded<N extends Name>(name: N): Option<Value> {
-    return Option.from(this._cascaded.get(name));
-  }
+  public specified<N extends Name>(name: N): Value<Style.Specified<N>> {
+    const {
+      options: { inherits },
+    } = Property.get(name);
 
-  public specified<N extends Name>(name: N): Value<Style.Specified<N>>;
-  public specified<N extends Name>(name: N): Value {
     return this.cascaded(name)
       .map((cascaded) => {
-        if (Keyword.isKeyword(cascaded.value)) {
-          switch (cascaded.value.value) {
+        const { value, source } = cascaded;
+
+        if (Keyword.isKeyword(value)) {
+          switch (value.value) {
+            // https://drafts.csswg.org/css-cascade/#initial
             case "initial":
               return this.initial(name);
 
+            // https://drafts.csswg.org/css-cascade/#inherit
             case "inherit":
-              return this.parent.computed(name);
+              return this.inherited(name);
+
+            // https://drafts.csswg.org/css-cascade/#inherit-initial
+            case "unset":
```

Comment:
By using a string as our guaranteed invalid and invalid at computed-value time values, we might run in a corner case if somebody uses `--foo: unset`.
OTOH, here also, I think this will not result in a different result ultimately (because "unset" would be invalid at computed-value time for any other property).

Reply:
`unset` is what declarations with guaranteed-invalid values will ultimately resolve to and so we just skip some needles, intermediary steps. It's a globally recognised defaulting keyword similar to `inherit` and `initial`.